### PR TITLE
Add CHANGELOG for AppConfiguration

### PR DESCRIPTION
**Pull request by alzimmermsft**
Adds a CHANGELOG that outlines that preview.1 was our initial release and the changes made between preview.1 and preview.2

### CODE REVIEW

File: sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Change Log azure-data-appconfiguration
+
+## Version 1.0.0-preview.2:
+
+For details on the Azure SDK for Java (August 2019 Preview) release refer to the [release announcement](https://aka.ms/azure-sdk-preview2-java).
+
+- Merged ConfigurationClientBuilder and ConfigurationAsyncClientBuilder into ConfigurationClientBuilder. Method to build each client were added
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md]: Should end bulleted lists with periods as part of the style guide.
https://docs.microsoft.com/en-ca/style-guide/punctuation/periods

File: sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+- ConfigurationClientBuilder was made instantiable, static builder method removed from ConfigurationClient and ConfigurationAsyncClient
+- Builder method credentials renamed to credential and serviceEndpoint to endpoint
+- Listing operations return PagedFlux and PagedIterable in their respective clients
+- Non-maximal overloads return response value instead of Response<T>
+- Asynchronous calls check subscriberContext for tracing context
+- Synchronous calls support passing tracing context in maximal overloads
+
```
Comment (conniey) [inline sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md]: unnecessary blank line.

File: sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md (hunk 1 above)
Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md]: Added periods

File: sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md (hunk 2 above)
Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md]: Removed blank line

File: sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+- ConfigurationClientBuilder was made instantiable, static builder method removed from ConfigurationClient and ConfigurationAsyncClient.
+- Builder method credentials renamed to credential and serviceEndpoint to endpoint.
+- Listing operations return PagedFlux and PagedIterable in their respective clients.
+- Non-maximal overloads return response value instead of Response<T>.
+- Asynchronous calls check subscriberContext for tracing context.
+- Synchronous calls support passing tracing context in maximal overloads.
+
+## Version 1.0.0-preview.1:
+
+Version 1.0.0-preview.1 is a preview of our efforts in creating a client library that is developer-friendly, idiomatic
```
Comment (sima-zhu) [inline sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md]: Do you think it is better to put the java spec on the top as it applies to all versions.

Comment (alzimmermsft) [reply sdk/appconfiguration/azure-data-appconfiguration/CHANGELOG.md]: For now I think it makes sense here as we could have different guidelines in the future.

**Pull request merged**